assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -124,6 +127,156 @@ export default {
+
+            get route() {
+              return "chat.channel";
+            }
+
+            get model() {
+              return {
+                chatChannel: this.channel,
+                channelId: this.channel.id,
+                channelTitle: this.channel.chatable.slug,
+              };
+            }
+
+            get title() {
+              return this.channel.title;
+            }
+
+            get text() {
+              return this.channel.title;
+            }
+
+            get prefixIcon() {
+              return "hashtag";
+            }
+
+            get prefixIconColor() {
+              return this.channel.chatable.color;
+            }
+
+            get prefixIconBadge() {
+              return this.channel.chatable.read_restricted && "lock";
+            }
+
+            get suffixIcon() {
+              return this.channel.unread_count > 0 && "circle";
+            }
+
+            get suffixCssClass() {
+              return this.channel.unread_mentions > 0 ? "urgent" : "unread";
+            }
+          };
+
+          const SidebarChatSection = class extends BaseCustomSidebarSection {
+            @tracked sectionLinks = A([]);
+
+            constructor({ sidebar }) {
+              super(...arguments);
+
+              this.sidebar = sidebar;
+              this.chatService = container.lookup("service:chat");
+              this.sidebar.appEvents.on(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.sidebar.appEvents.on(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+              this.currentUser = getOwner(this).lookup("current-user:main");
+            }
+
+            willDestroy() {
+              this.sidebar.appEvents.off(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.sidebar.appEvents.off(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+            }
+
+            @bind
+            _refreshChannels() {
+              const newSectionLinks = [];
+              this.chatService.getChannels().then((channels) => {
+                channels.publicChannels.forEach((channel) => {
+                  const trackingState =
+                    this.currentUser.chat_channel_tracking_state[channel.id];
+                  if (channel === this.chatService.activeChannel) {
+                    channel.unread_count = 0;
+                    channel.unread_mentions = 0;
+                  } else {
+                    channel.unread_count = trackingState.unread_count;
+                    channel.unread_mentions = trackingState.unread_mentions;
+                  }
+                  newSectionLinks.push(new SidebarChatSectionLink({ channel }));
+                });
+              });
+              this.sectionLinks = newSectionLinks;
+            }
+
```

Comment:
Not sure name/title/text are descriptive enough, have a hard time understanding what it would be from here. Also why are they all the same "chat.chat_channels" ?

Reply:
I tried to follow this convention:
https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/templates/components/sidebar/topics-section.hbs#L12

- name - component identifier which just setup CSS class https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/templates/components/sidebar/section-link.hbs#L3
- title - text for link
- text - text for "alt" attribute

You are right that name was wrong, I changed it to `chat-channels`

I can rename those attributes for example:
name -> identifier
title
text -> alternativeText

In that case, I would like to change core to have some consistency. Does it sound like a good idea to you?

Reply:
If it's used for alt attribute, I would just call it `alt`. Im fine with this but I don't know everything about sidebar.

Reply:
I wouldn't worry too much about this because the parent class is document and anyone that implements the parent's class interface will have to read the docs.